genai-perf/genai_perf/metrics/telemetry_metrics.py

```diff
@@ -58,19 +58,19 @@ class TelemetryMetrics:
 
     def __init__(
         self,
-        gpu_power_usage: Dict[str, List[float]] = defaultdict(list),
-        gpu_power_limit: Dict[str, List[float]] = defaultdict(list),
-        energy_consumption: Dict[str, List[float]] = defaultdict(list),
-        gpu_utilization: Dict[str, List[float]] = defaultdict(list),
-        total_gpu_memory: Dict[str, List[float]] = defaultdict(list),
-        gpu_memory_used: Dict[str, List[float]] = defaultdict(list),
-    ) -> None:
-        self.gpu_power_usage = gpu_power_usage
-        self.gpu_power_limit = gpu_power_limit
-        self.energy_consumption = energy_consumption
-        self.gpu_utilization = gpu_utilization
-        self.total_gpu_memory = total_gpu_memory
-        self.gpu_memory_used = gpu_memory_used
+        gpu_power_usage: Optional[Dict[str, List[float]]] = None,
+        gpu_power_limit: Optional[Dict[str, List[float]]] = None,
+        energy_consumption: Optional[Dict[str, List[float]]] = None,
+        gpu_utilization: Optional[Dict[str, List[float]]] = None,
+        total_gpu_memory: Optional[Dict[str, List[float]]] = None,
+        gpu_memory_used: Optional[Dict[str, List[float]]] = None,
+    ):
+        self.gpu_power_usage = defaultdict(list, gpu_power_usage or {})
```

Comment:
Using default dictionaries in initialization parameters is dangerous if they later get updated. The state is shared among default dictionaries/lsits.

The new unit tests revealed this. One unit test used the default value and appended to the dictionary in the Telemetry metrics, then the next unexpectedly had metric values when initialized with default values. This is the more standard approach when a parameter can have a default dictionary to avoid sharing state.